Userland/Services/WindowServer/main.cpp

```diff
@@ -96,6 +96,20 @@ int main(int, char**)
             wm_config->read_num_entry(group_name, "Width", 1024 / scale),
             wm_config->read_num_entry("Screen", "Height", 768 / scale)
         };
+
+        // Check if the screen would be overlapping with another one
+        if (WindowServer::Screen::for_each([&](auto& screen) {
+                if (screen.rect().intersects(virtual_rect)) {
+                    dbgln("Screen {} rect {} overlaps with screen {} rect {}: Ignoring configuration", screen.index(), screen.rect(), screen_index, virtual_rect);
+                    return IterationDecision::Break;
+                }
+                return IterationDecision::Continue;
+            })
+            == IterationDecision::Break) {
```

Comment:
I'm not a fan of the formatting there. I think it would look cleaner with a temporary :
```suggestion
        auto overlap_detection_result = WindowServer::Screen::for_each([&](auto& screen) {
                if (screen.rect().intersects(virtual_rect)) {
                    dbgln("Screen {} rect {} overlaps with screen {} rect {}: Ignoring configuration", screen.index(), screen.rect(), screen_index, virtual_rect);
                    return IterationDecision::Break;
                }
                return IterationDecision::Continue;
            });
        if (overlap_detection_result == IterationDecision::Break) {
```

Reply:
This is actually an interesting find. I had meant to move this code into ScreenLayout.ipp and ended up just deleting it at some point, without adding it to ScreenLayout.ipp... where it is also a lot cleaner/nicer.